internal/git/gitsync_processor.go

```diff
@@ -42,17 +44,14 @@ type GitSyncProcessor struct {
 	clusterName string
 }
 
-func watchRepo(ctx context.Context, repo *v1.RepositoryPath, _ /* namespace */ string) error {
-	logger := logging.FromContext(ctx)
-
-	r, err := git.Clone(memory.NewStorage(), nil, &git.CloneOptions{
-		URL:          repo.RepoUrl,
-		SingleBranch: false,
+func cloneRepo(repo *v1.RepositoryPath) (*git.Repository, error) {
+	return git.Clone(memory.NewStorage(), nil, &git.CloneOptions{
+		URL: repo.RepoUrl,
 	})
-	if err != nil {
-		logger.Errorw("error cloning the repository", "err", err)
-		return err
-	}
+}
+
+func watchRepo(ctx context.Context, r *git.Repository, gitSync *v1.GitSync, k8Client client.Client, repo *v1.RepositoryPath, _ /* namespace */ string) error {
+	logger := logging.FromContext(ctx)
```

Comment:
can we set this logger to include GitSync name and RepositoryPath name in it using `logging.FromContext(ctx).With(...)`? Then we won't need to add it to each logger line below